src/domain/Processors/VehicleProcessor.ts

```diff
@@ -150,7 +150,7 @@ export abstract class VehicleProcessor<T extends Vehicle> {
       return vehiclesFromDB[0];
     }
     const filteredRecords = vehiclesFromDB.filter((record) => record.techRecord.find(findFn));
-    if (filteredRecords.length > 1) {
+    if (filteredRecords.length !== 1) {
```

Comment:
Is this in case filtered records is less than one?

Reply:
Exactly, I managed to make it get a 0 on the filter and that broke some stuff, thought this is closer to what we actually wanted.